### PR TITLE
Send more additional information to listenbrainz

### PR DESCRIPTION
**Pull request by badlandspray**
As per API [spec](https://listenbrainz.readthedocs.io/en/latest/users/json.html#payload-json-details), send `duration` and `spotify_id` in `additional_info`

### CODE REVIEW

**Comment by jaccarmac** [comment]
I'm not a user of the Listenbrainz feature and largely unfamiliar with the core of the codebase, but this looks pretty reasonable.

**Comment by inverse** [comment]
Thanks for contributing @badlandspray and thanks for the review @jaccarmac 

We're lucky enough that we have engineers from Listenbrainz who are active in this community.

@MonkeyDo Would you be able to check this one?

**Comment by MonkeyDo** [comment]
Absolutely! I'm AFK this coming week but count on me once I'm back.

**Comment by badlandspray** [comment]
@MonkeyDo Would you able to review it now?

File: src/core/background/scrobbler/listenbrainz-scrobbler.js
Comment (inverse) [inline src/core/background/scrobbler/listenbrainz-scrobbler.js:288]: This would be something like

```
spotify:track:0VfQeEvwYx4NXuGXkbWgUb
```

@MonkeyDo is that correct? 

The spec outlines

>  The Spotify track URL associated with this recording. e.g.: http://open.spotify.com/track/1rrgWMXGCGHru5bIRxGFV0

File: src/core/background/scrobbler/listenbrainz-scrobbler.js
Comment (MonkeyDo) [inline src/core/background/scrobbler/listenbrainz-scrobbler.js:292]: We now [clarified](https://listenbrainz.readthedocs.io/en/latest/users/json.html#payload-json-details) that we accept either `duration` or `duration_ms`, the former for durations in seconds and the latter (which I would recommend) for durations in milliseconds

File: src/core/background/scrobbler/listenbrainz-scrobbler.js
Comment (MonkeyDo) [inline src/core/background/scrobbler/listenbrainz-scrobbler.js:288]: We have improved the metadata section of the listen and made them a bit more useful (see reference [here](https://listenbrainz.readthedocs.io/en/latest/users/json.html#brainzplayer-on-the-listenbrainz-website-playing-a-video-from-spotify))

The field `track_metadata.additional_info.origin_url` should also be added, in this case with the same value as the `spotify_id` field

**Review by MonkeyDo — COMMENTED** [review]
Sorry about the long delay!
We've been finalizing some changes to the suggested structure of the track metadata payload (https://github.com/metabrainz/listenbrainz-server/pull/1607, https://github.com/metabrainz/listenbrainz-server/pull/1832), I figured it would be annoying to validate the PR to then propose some extra changes !

File: src/core/background/scrobbler/listenbrainz-scrobbler.js
Comment (MonkeyDo) [reply src/core/background/scrobbler/listenbrainz-scrobbler.js:288]: Indeed the `spotify_id` and the `origin_url` fields should both be a fully-qualified rather than the `spotify:track:…` URN

File: src/core/background/scrobbler/listenbrainz-scrobbler.js
Comment (inverse) [reply src/core/background/scrobbler/listenbrainz-scrobbler.js:292]: What's the rational for using `ms`? internally the connector uses seconds for duration so we would have the same precision if we multiplied this value up?

**Comment by badlandspray** [comment]
Looks like the release been cut, I was hoping to get this in the release. Have to wait for one more month for the next release?

**Comment by inverse** [comment]
@badlandspray You hopefully won't have to wait that long - releases are pretty automated so would be just a case of getting an approval and running a few commands.

File: src/core/background/scrobbler/listenbrainz-scrobbler.js
Comment (MonkeyDo) [reply src/core/background/scrobbler/listenbrainz-scrobbler.js:292]: In that case `duration` is the correct one, sorry if I wasn't clear.
I just wasn't sure of the internal implementation and wanted to make sure the right field is used.

**Review by inverse — APPROVED** [review]

**Pull request merged**
